R/DownloadModule.R

```diff
@@ -322,11 +318,301 @@ reporter_download_inputs <- function(rmd_yaml_args, rmd_output, showrcode, sessi
+  lines <- readLines(input_rmd)
+
+  new_lines <- c()
+  skip_chunk <- FALSE  # Track if we are inside an 'object' chunk
+
+  for (line in lines) {
+    # Start removing 'object' chunks
+    if (stringr::str_detect(line, "^```\\{r object_")) {
+      skip_chunk <- TRUE
+    }
+
+    # Stop skipping when the chunk ends
+    if (stringr::str_detect(line, "^```$") && skip_chunk) {
+      skip_chunk <- FALSE
+      next  # Skip adding this line
+    }
+
+    if (!skip_chunk) {
+      # Remove eval=FALSE from any R chunk
+      line <- stringr::str_replace(line, ",?\\s*eval=FALSE", "")
+
+      # Clean up extra commas in chunk headers
+      line <- stringr::str_replace(line, "\\{r,\\s*,", "{r,")
+      line <- stringr::str_replace(line, "\\{r,", "{r")  # Remove leftover commas
+
+      new_lines <- c(new_lines, line)
+    }
+  }
+
+  writeLines(new_lines, input_rmd)
+}
+
+report_render_Rmd <- function(blocks, yaml_header, global_knitr = getOption("teal.reporter.global_knitr"), output_dir) {
+  checkmate::assert_list(
+    blocks,
+    c("TextBlock", "PictureBlock", "NewpageBlock", "TableBlock", "RcodeBlock", "HTMLBlock", "character",
+      "gg", "rtables", "TableTree", "ElementaryTable", "rlisting", "data.frame")
+  )
+  checkmate::assert_subset(names(global_knitr), names(knitr::opts_chunk$get()))
+  if (missing(yaml_header)) {
+    yaml_header <- md_header(yaml::as.yaml(list(title = "Report")))
+  }
+
+  report_type <- get_yaml_field(yaml_header, "output")
+
+  parsed_global_knitr <- sprintf(
+    "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n%s\n```\n",
+    capture.output(dput(global_knitr)),
+    if (identical(report_type, "powerpoint_presentation")) {
+      format_code_block_function <- quote(
+        code_block <- function(code_text) {
+          df <- data.frame(code_text)
+          ft <- flextable::flextable(df)
+          ft <- flextable::delete_part(ft, part = "header")
+          ft <- flextable::autofit(ft, add_h = 0)
+          ft <- flextable::fontsize(ft, size = 7, part = "body")
+          ft <- flextable::bg(x = ft, bg = "lightgrey")
+          ft <- flextable::border_outer(ft)
+          if (flextable::flextable_dim(ft)$widths > 8) {
+            ft <- flextable::width(ft, width = 8)
+          }
+          ft
+        }
+      )
+      paste(deparse(format_code_block_function), collapse = "\n")
+    } else {
+      ""
+    }
+  )
+
+  parsed_blocks <- paste(
+    unlist(
+      lapply(blocks, function(b) block_to_md(b, output_dir = output_dir, report_type = report_type))
+    ),
+    collapse = "\n\n"
+  )
+
+  rmd_text <- paste0(yaml_header, "\n", parsed_global_knitr, "\n", parsed_blocks, "\n")
+  tmp <- tempfile(fileext = ".Rmd")
+  input_path <- file.path(
+    output_dir,
+    sprintf("input_%s.Rmd", gsub("[.]", "", format(Sys.time(), "%Y%m%d%H%M%OS3")))
+  )
+  cat(rmd_text, file = input_path)
+  input_path
+}
+
+#' @keywords internal
+block_to_md <- function(block, output_dir, report_type, ...) {
```

Comment:
In your implementation block is turned into Rmd not md. 
```suggestion
block_to_rmd <- function(block, output_dir, report_type, ...) {
```

Reply:
9ae2e46aeaa77c72ea455e9c6767ca305cfa7e6e